### PR TITLE
🌱 Add dependabot config and workflow

### PR DESCRIPTION
**Pull request by kashifest**
This PR adds the dependabot config and workflow which will handle go module updates and other bumps.

### CODE REVIEW

**Review by Sunnatillo — COMMENTED** [review]
LGTM

File: .github/workflows/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+    - name: Calculate go version
+      id: vars
+      run: echo "go_version=$(make go-version)" >> $GITHUB_OUTPUT
+    - name: Set up Go
+      uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # tag=v4.0.1
```
Comment (tuminoid) [inline .github/workflows/dependabot.yml:26]: Make the comments consistent, as we saw in other PR, dependabot trips on these sometimes.
```suggestion
      uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
```

File: .github/workflows/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      with:
+        go-version: ${{ steps.vars.outputs.go_version }}
+    - uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # tag=v3.3.1
```
Comment (tuminoid) [inline .github/workflows/dependabot.yml:29]:
```suggestion
    - uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
```

File: .github/workflows/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Update all modules
+      run: make modules
+    - name: Update generated code
+      run: make generate
+    - uses: EndBug/add-and-commit@1bad3abcf0d6ec49a5857d124b0bfb52dc7bb081 # tag=v9.1.3
```
Comment (tuminoid) [inline .github/workflows/dependabot.yml:42]:
```suggestion
    - uses: EndBug/add-and-commit@1bad3abcf0d6ec49a5857d124b0bfb52dc7bb081 # v9.1.3
```

**Review by tuminoid — COMMENTED** [review]
Please fix typo in title as well.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    schedule:
+      interval: "weekly"
+    ignore:
+      # Ignore controller-runtime as its upgraded manually.
+      - dependency-name: "sigs.k8s.io/controller-runtime"
+      # Ignore k8s and its transitives modules as they are upgraded manually
+      # together with controller-runtime.
+      - dependency-name: "k8s.io/*"
+      - dependency-name: "go.etcd.io/*"
+      - dependency-name: "google.golang.org/grpc"
```
[the remark this answers is not in the document]
Comment (kashifest) [reply .github/dependabot.yml:26]: Yes, actually it is in gosum only. May be I should remove it.

[the remark this answers is not in the document]
Comment (kashifest) [reply .github/dependabot.yml]: Done

**Review by furkatgofurov7 — APPROVED** [review]

File: .github/dependabot.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kashifest) [reply .github/dependabot.yml]: You need to reapprove @furkatgofurov7

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Pull request merged**